### PR TITLE
add connectivity tab to line and transfo modification form

### PR DESCRIPTION
**Pull request by ghazwarhili**

### CODE REVIEW

**Review by EtienneLt — APPROVED** [review]

File: src/components/dialogs/network-modifications/line/characteristics-pane/line-connectivity-pane.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    const connectivity1Field = (
+        <ConnectivityForm
+            id={CONNECTIVITY_1}
+            studyUuid={studyUuid}
+            currentNode={currentNode}
+            voltageLevelOptions={voltageLevelOptions}
+            withPosition={true}
+            isEquipmentModification={isModification}
+            previousValues={previousValues}
```
Comment (thangqp) [inline src/components/dialogs/network-modifications/line/characteristics-pane/line-connectivity-pane.tsx:40]:
```suggestion
            previousValues={{
                connectablePosition: previousValues.connectablePosition1,
                terminalConnected: previousValues.terminal1Connected,
            }}
```

File: src/components/dialogs/network-modifications/line/characteristics-pane/line-connectivity-pane.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    const connectivity2Field = (
+        <ConnectivityForm
+            id={CONNECTIVITY_2}
+            studyUuid={studyUuid}
+            currentNode={currentNode}
+            voltageLevelOptions={voltageLevelOptions}
+            withPosition={true}
+            isEquipmentModification={isModification}
+            previousValues={previousValues}
```
Comment (thangqp) [inline src/components/dialogs/network-modifications/line/characteristics-pane/line-connectivity-pane.tsx:52]:
```suggestion
            previousValues={{
                connectablePosition: previousValues.connectablePosition12,
                terminalConnected: previousValues.terminal2Connected,
            }}
```

File: src/components/dialogs/network-modifications/line/characteristics-pane/line-connectivity-pane.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import { Grid } from '@mui/material';
+import { gridItem, GridSection } from '../../../dialogUtils';
+import { ConnectivityForm } from '../../../connectivity/connectivity-form';
+import { CONNECTIVITY_1, CONNECTIVITY_2 } from 'components/utils/field-constants';
+import React, { FunctionComponent } from 'react';
+import useVoltageLevelsListInfos from '../../../../../hooks/use-voltage-levels-list-infos';
+import { CurrentTreeNode } from '../../../../../redux/reducer';
+import { UUID } from 'crypto';
+
+interface LineConnectivityPaneProps {
```
Comment (thangqp) [inline src/components/dialogs/network-modifications/line/characteristics-pane/line-connectivity-pane.tsx:17]: You can rename this component to LineConnectivityForm and move to dialogs/connectivity. This is a shared component by 2TW and Line so it is better to move this component in a separated folder

File: src/components/dialogs/network-modifications/line/modification/line-modification-dialog.jsx [hunk 4]
```diff
@@ -98,12 +114,14 @@ const LineModificationDialog = ({
...
     const [lineToModify, setLineToModify] = useState(null);
-    const [tabIndex, setTabIndex] = useState(LineCreationDialogTab.CHARACTERISTICS_TAB);
+    const [tabIndex, setTabIndex] = useState(LineModificationDialogTab.CONNECTIVITY_TAB);
     const [isOpenLineTypesCatalogDialog, setOpenLineTypesCatalogDialog] = useState(false);
 
     const emptyFormData = useMemo(
         () => ({
             [EQUIPMENT_NAME]: '',
+            ...getConnectivityWithPositionEmptyFormData(true, CONNECTIVITY_1),
+            ...getConnectivityWithPositionEmptyFormData(true, CONNECTIVITY_2),
```
Comment (thangqp) [inline src/components/dialogs/network-modifications/line/modification/line-modification-dialog.jsx:124]: The root of each tab should be coherent to the tab name, i.e. connectivity. If not, onValidationError does not work, although in this modification form, it is not evident to produce a situation in which the tab connectivity not valid and the tab header should be in 'red'.

File: src/components/dialogs/network-modifications/line/characteristics-pane/line-connectivity-pane.tsx (hunk 2 above)
Comment (ghazwarhili) [reply src/components/dialogs/network-modifications/line/characteristics-pane/line-connectivity-pane.tsx]: done

File: src/components/dialogs/network-modifications/line/modification/line-modification-dialog.jsx [hunk 5]
```diff
@@ -135,6 +163,8 @@ const LineModificationDialog = ({
             }
             reset({
                 [EQUIPMENT_NAME]: line.equipmentName?.value ?? '',
+                [CONNECTIVITY]: getConnectivityFormData(createConnectivityData(line, 1), CONNECTIVITY_1),
+                [CONNECTIVITY]: getConnectivityFormData(createConnectivityData(line, 2), CONNECTIVITY_2),
```
Comment (thangqp) [inline src/components/dialogs/network-modifications/line/modification/line-modification-dialog.jsx:167]:
```suggestion
                [CONNECTIVITY]: {
                    ...getConnectivityFormData(createConnectivityData(line, 1), CONNECTIVITY_1),
                    ...getConnectivityFormData(createConnectivityData(line, 2), CONNECTIVITY_2),
                },
```

Comment (ghazwarhili) [reply src/components/dialogs/network-modifications/line/modification/line-modification-dialog.jsx]: done

File: src/components/dialogs/network-modifications/line/modification/line-modification-dialog.jsx [hunk 6]
```diff
@@ -223,6 +255,18 @@ const LineModificationDialog = ({
...
+                connectivity2[VOLTAGE_LEVEL]?.id,
+                connectivity2[BUS_OR_BUSBAR_SECTION]?.id,
+                sanitizeString(connectivity1[CONNECTION_NAME]),
+                sanitizeString(connectivity2[CONNECTION_NAME]),
+                connectivity1[CONNECTION_DIRECTION] ?? UNDEFINED_CONNECTION_DIRECTION,
+                connectivity2[CONNECTION_DIRECTION] ?? UNDEFINED_CONNECTION_DIRECTION,
+                connectivity1[CONNECTION_POSITION] ?? null,
+                connectivity2[CONNECTION_POSITION] ?? null,
+                connectivity1[CONNECTED] ?? null,
+                connectivity2[CONNECTED] ?? null,
```
Comment (thangqp) [inline src/components/dialogs/network-modifications/line/modification/line-modification-dialog.jsx:262]:
```suggestion
                connectivity1[CONNECTION_DIRECTION],
                connectivity2[CONNECTION_DIRECTION],
                connectivity1[CONNECTION_POSITION],
                connectivity2[CONNECTION_POSITION],
                connectivity1[CONNECTED],
                connectivity2[CONNECTED],
```
Why you need set default value for these fields?

File: src/components/dialogs/network-modifications/two-windings-transformer/modification/two-windings-transformer-modification-dialog.jsx [hunk 7]
```diff
@@ -474,6 +506,18 @@ const TwoWindingsTransformerModificationDialog = ({
...
+                connectivity2[VOLTAGE_LEVEL]?.id,
+                connectivity2[BUS_OR_BUSBAR_SECTION]?.id,
+                sanitizeString(connectivity1[CONNECTION_NAME]),
+                sanitizeString(connectivity2[CONNECTION_NAME]),
+                connectivity1[CONNECTION_DIRECTION] ?? UNDEFINED_CONNECTION_DIRECTION,
+                connectivity2[CONNECTION_DIRECTION] ?? UNDEFINED_CONNECTION_DIRECTION,
+                connectivity1[CONNECTION_POSITION] ?? null,
+                connectivity2[CONNECTION_POSITION] ?? null,
+                connectivity1[CONNECTED] ?? null,
+                connectivity2[CONNECTED] ?? null,
```
Comment (thangqp) [inline src/components/dialogs/network-modifications/two-windings-transformer/modification/two-windings-transformer-modification-dialog.jsx:513]: Same question as in line, why we need set default values for these fields?

File: src/components/dialogs/network-modifications/line/modification/line-modification-dialog.jsx (hunk 6 above)
Comment (thangqp) [reply src/components/dialogs/network-modifications/line/modification/line-modification-dialog.jsx:262]: In addition, UNDEFINED_CONNECTION_DIRECTION as a default value is not coherent to null that you set in createConnectivityData

**Review by thangqp — APPROVED** [review]

**Pull request merged**